software/database/src/main/java/org/apache/brooklyn/entity/database/postgresql/PostgreSqlSshDriver.java

```diff
@@ -328,9 +340,39 @@ private void initializeNewDatabase() {
                         " --command="+ createUserCommand),
                 sudoAsUser("postgres", getInstallDir() + "/bin/psql -p " + entity.getAttribute(PostgreSqlNode.POSTGRESQL_PORT) + 
                                 " --command="+ createDatabaseCommand),
+                createRolesAdditionalCommand,
                 callPgctl("stop", true))
                 .failOnNonZeroResultCode().execute();
     }
+
+    private String buildCreateRolesQuery() {
+        Map<String, Map> roles = entity.getConfig(PostgreSqlNode.ROLES);
+        String createRolesQuery = "\"";
+
+        for (Map.Entry role: roles.entrySet()) {
+            createRolesQuery = createRolesQuery.concat(String.format("CREATE ROLE %s", role.getKey()));
```

Comment:
Here and below, you shouldn't create SQL statements using string concatenation, that leaves you wide open to SQL injection attacks.  Instead use a prepared statement with appropriate bind variables, if at all possible.